share/functions/__fish_print_pipestatus.fish

```diff
@@ -1,19 +1,35 @@
 function __fish_print_pipestatus --description "Print pipestatus for prompt"
-    # maybe these could be in global variables similar to __fish_color_status to allow
-    # users to set the variables to modify the braces/separator/color used
-    set -l left_brace $argv[1]
-    set -l right_brace $argv[2]
-    set -l separator $argv[3]
-    set -l brace_sep_color $argv[4]
-    set -l status_color $argv[5]
+
+    # allow users to customize these variables
+    if not set -q __fish_prompt_pipestatus_left_brace
+        set -g __fish_prompt_pipestatus_left_brace $argv[1]
+    end
+
+    if not set -q __fish_prompt_pipestatus_right_brace
+        set -g __fish_prompt_pipestatus_right_brace $argv[2]
+    end
+
+    if not set -q __fish_prompt_pipestatus_separator
+        set -g __fish_prompt_pipestatus_separator $argv[3]
```

Comment:
How about a default of '|' if there is no $argv[3]?

Reply:
I am not sure how to accomplish that as the arguments are positional.

Reply:
```fish
if not set -q argv[3]
    set -g __fish_prompt_pipestatus_separator '|'
else
    set -g __fish_prompt_pipestatus_separator $argv[3]
end
```

?

Reply:
One caveat here is that "local" variables are local to the block, so if you do 

```fish
if not set -q argv[3]
    set -l separator '|'
```

then $separator will not be defined after the `if`-block ends. That means you need to define the variable before:

```fish
set -l separator
if not set -q argv[3]
   set separator '|'
```

or use `and` and `or` instead of `if`/`else` because they don't introduce a block.

Reply:
I experimented with this, not sure how to make it work. If `argv[3]` is omitted then `argv[4]` becomes `argv[3]` (no holes in arrays). So `argv[3]` is always set. If I call `__fish_print_pipestatus` with `argv[3]` set to `''` (empty string) then it is still considered set.
I can modify the `if not set -q argv[3]` to `if test -z $argv[3]`?

Reply:
Noo... that would require passing `''`.

Instead, list the variables in order of importance, and later ones that aren't passed fall back to their defaults.

I.e. if you read "separator color1", then it's just not possible to pass a color without also passing the separator. That's okay. If two arguments are passed, they're used. If only one argument is passed, that's the separator and the color defaults to e.g. blue. If no argument is passed, separator defaults to `|` and color defaults to blue.

The alternative is to allow passing "--separator=SEP" - see [argparse](http://fishshell.com/docs/current/commands.html#argparse) for a way to do that easily. But in this case it's probably not worth it. The function doesn't have to be hyperversatile in every single way. It's okay if a user with very specific needs uses `funced`, or if they need to specify the default separator.

Reply:
The separator cannot be last as the actual `$pipestatus` array needs to be last as it is variable in size. If I add separator after that and it is optional I will lose the ability to accurately determine what `$pipestatus` values are. It is called like this:
```
__fish_print_pipestatus "[" "] " "|" (set_color yellow) (set_color --bold yellow) $last_pipestatus
```
The alternative would be to use argparse and provide defaults for all of braces, colors and separator. Is that desirable?

Reply:
>The separator cannot be last as the actual $pipestatus array needs to be last as it is variable in size

Oooh, sorry, I missed that!

>The alternative would be to use argparse and provide defaults for all of braces, colors and separator. Is that desirable?

No, keep it as it is.